src/App.jsx

```diff
@@ -14,33 +16,98 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button type="button" className="button is-info is-light">
-        Sort alphabetically
-      </button>
+const SORT_FIELD_ALPHABET = 'Alphabet';
+const SORT_FIELD_LENGTH = 'Length';
 
-      <button type="button" className="button is-success is-light">
-        Sort by length
-      </button>
+function prepareGoods(goods, { type, reverse }) {
+  const preparedGoods = [...goods];
 
-      <button type="button" className="button is-warning is-light">
-        Reverse
-      </button>
+  preparedGoods.sort((good1, good2) => {
+    switch (type) {
+      case SORT_FIELD_ALPHABET:
+        return good1.localeCompare(good2);
+      case SORT_FIELD_LENGTH:
+        return good1.length - good2.length;
+    }
 
-      <button type="button" className="button is-danger is-light">
-        Reset
-      </button>
-    </div>
+    return preparedGoods;
+  });
+
+  if (reverse) {
+    preparedGoods.reverse();
+  }
+
+  return preparedGoods;
+}
+
+export const App = () => {
+  const [sortQuery, setSortQuery] = useState({
+    type: '',
+    reverse: false,
+  });
+  const sortedGoods = prepareGoods(goodsFromServer, sortQuery);
+
+  const setQueryValue = (queryType, queryValue) => {
+    if (queryValue) {
+      setSortQuery(prev => ({ ...prev, [queryType]: queryValue }));
+    } else {
+      setSortQuery(prev => ({ ...prev, [queryType]: !prev[queryType] }));
+    }
+  };
```

Comment:
I think that if you want to manage the state like that the better option would be the useReducer hook.

Reply:
Better update two different states, no need to create one universal handler for them. Readability is everything